test/Microsoft.Framework.Runtime.Tests/JsonBufferFacts.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.Framework.Runtime.Json.Tests
+{
+    public class JsonBufferFacts
+    {
+        [Fact]
+        public void BasicReadToken()
+        {
+            var content = @"
+            {
+                ""key1"": ""value1"",
+                ""key2"": 99,
+                ""key3"": true,
+                ""key4"": [""str1"", ""str2"", ""str3""],
+                ""key5"": {
+                    ""subkey1"": ""subvalue1"",
+                    ""subkey2"": [1, 2]
+                },
+                ""key6"": null
+            }";
+
+            using (var stream = CreateTextReader(content))
+            {
+                var buffer = new JsonBuffer(stream);
+                Assert.NotNull(buffer);
+
+                while (buffer.Read().Type != JsonTokenType.EOL)
+                    ;
+            }
+        }
+
+        [Fact]
+        public void ReadTokenThroughLockFileSample()
+        {
+            using (var stream = File.OpenRead(".\\TestSample\\project.lock.sample"))
+            {
+                var reader = new StreamReader(stream);
+                var buffer = new JsonBuffer(reader);
+                Assert.NotNull(buffer);
+
+                while (buffer.Read().Type != JsonTokenType.EOL)
```

Comment:
Um... this test could pass even when things go wrong. If the buffer returns EOL after the first read, this test passes.